sql/database.go

```diff
@@ -136,7 +136,19 @@ func (db *Database) Tx(ctx context.Context) (*Tx, error) {
 		return nil, ErrNoConnection
 	}
 	tx := &Tx{db: db, conn: conn}
-	return tx, tx.begin()
+	return tx, tx.deferred()
+}
+
+// TxImmediate creates immediate transaction.
+//
+// IMMEDIATE cause the database connection to start a new write immediately, without waiting // for a write statement. The BEGIN IMMEDIATE might fail with SQLITE_BUSY if another write //// transaction is already active on another database connection.
```

Comment:
fix this comment.

do you mean BEGIN might fail with SQLITE_BUSY?

Reply:
noticed this as well

yes, so if there is another write in progress database will wait sufficiently long before terminating tx with SQLITE_BUSY. but when it is initialized tx with read statement, and then upgrades to write - it doesn't wait and terminates immediately